concourse/util.py

```diff
@@ -46,130 +42,114 @@ def list_github_resources(
 
 
 def sync_webhooks(
-    github_cfg: GithubConfig,
-    concourse_cfg: ConcourseConfig,
-    job_mapping: JobMapping,
-    concourse_team_credentials: ConcourseTeamCredentials,
-    concourse_pipelines: [str]=None,
-    concourse_verify_ssl: bool=True,
+    webhook_dispatcher_cfg: WebhookDispatcherConfig,
+    # TODO: There is currently nothing explicitly linking a whd-cfg and a whd-deployment-cfg
+    webhook_dispatcher_deployment_cfg: WebhookDispatcherDeploymentConfig,
 ):
-    concourse_team = concourse_team_credentials.teamname()
+    cfg_factory = ctx().cfg_factory()
 
-    github_resources = list_github_resources(
-        concourse_cfg=concourse_cfg,
-        concourse_team=concourse_team,
-        concourse_pipelines=concourse_pipelines,
-        github_url=github_cfg.http_url(),
-    )
-    # group by repositories
-    path_to_resources = {}
-    allowed_github_orgs = [
-        github_org.org_name() for github_org in job_mapping.github_organisations()
-    ]
-    for gh_res in github_resources:
-        repo_path = gh_res.github_source().repo_path()
-        if gh_res.github_source().parse_organisation() not in allowed_github_orgs:
-            continue
-        if repo_path not in path_to_resources:
-            path_to_resources[repo_path] = [gh_res]
-        else:
-            path_to_resources[repo_path].append(gh_res)
-
-    github_obj = _create_github_api_object(github_cfg=github_cfg)
-
-    webhook_syncer = github.webhook.GithubWebHookSyncer(github_obj)
+    concourse_cfg_names = webhook_dispatcher_cfg.concourse_config_names()
+    concourse_cfgs = map(cfg_factory.concourse, concourse_cfg_names)
     failed_hooks = 0
 
-    for repo, resources in path_to_resources.items():
-        try:
-            _sync_webhook(
-                resources=resources,
-                webhook_syncer=webhook_syncer,
-                job_mapping_name=job_mapping.name(),
-                concourse_cfg=concourse_cfg,
-                skip_ssl_validation=not concourse_verify_ssl
-            )
-        except Exception as e:
-            failed_hooks += 1
-            warning(f'repo: {repo} - error: {e}')
+    # define filter function here - it does not change for a given webhook_dispatcher_cfg
+    def url_filter(url):
+        parsed_url = parse_qs(urlparse(url).query)
+        whd_id = parsed_url.get(
+            github.webhook.WebhookQueryAttributes.WHD_ID_ATTRIBUTE_NAME
+        )
+        # consider an url for removal iff it contains parameter
+        # 'whd_id' matching given whd_id
+        return (
```

Comment:
for readability reaons, assign the value to a variable (with an expressive name, such as "should_remove" or at least "matches")